base/compiler/ssair/inlining.jl

```diff
@@ -1169,8 +1169,7 @@ function early_inline_special_case(ir::IRCode, s::Signature, e::Expr, params::Pa
             val = etype.val
             is_inlineable_constant(val) || return nothing
             if isa(f, IntrinsicFunction)
-                if is_pure_intrinsic_infer(f) &&
-                    (intrinsic_nothrow(f) || intrinsic_nothrow(f, atypes[2:end]))
+                if is_pure_intrinsic_infer(f) && intrinsic_nothrow(f, atypes[2:end])
```

Comment:
Do we need to handle Vararg early in this function and bail?

Reply:
No, vararg is handled earlier in inlining and it refuses to construct a `Signature` object for such cases.